ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/auth/OngoingStreamAuthIT.scala

```diff
@@ -0,0 +1,149 @@
+import com.daml.platform.sandbox.services.SubmitAndWaitDummyCommandHelpers
+import io.grpc.stub.StreamObserver
+import io.grpc.{Status, StatusRuntimeException}
+import com.daml.ledger.api.v1.admin.{user_management_service => user_management_service_proto}
+
+import scala.concurrent.{Future, Promise}
+
+final class OngoingStreamAuthIT
+    extends ServiceCallAuthTests
+    with SubmitAndWaitDummyCommandHelpers
+    with ErrorsAssertions {
+
+  private val UserManagementCacheExpiryInSeconds = 1
+
+  override protected def config: SandboxConfig = super.config.withUserManagementConfig(
+    _.copy(cacheExpiryAfterWriteInSeconds = UserManagementCacheExpiryInSeconds)
+  )
+
+  override def serviceCallName: String = ""
+
+  override protected def serviceCallWithToken(token: Option[String]): Future[Any] = ???
+
+  private val testId = UUID.randomUUID().toString
+
+  it should "abort an ongoing stream after user state has changed" in {
+    val partyAlice = "alice-party"
+    val userIdAlice = testId + "-alice"
+    val receivedTransactionsCount = new AtomicInteger(0)
+    val transactionStreamAbortedPromise = Promise[Throwable]()
+
+    def observeTransactionsStream(
+        token: Option[String],
+        party: String,
+    ): StreamObserver[GetTransactionsResponse] = {
+      val observer = new StreamObserver[GetTransactionsResponse] {
+        override def onNext(value: GetTransactionsResponse): Unit = {
+          val _ = receivedTransactionsCount.incrementAndGet()
+        }
+
+        override def onError(t: Throwable): Unit = {
+          val _ = transactionStreamAbortedPromise.trySuccess(t)
+        }
+
+        override def onCompleted(): Unit = ()
+      }
+      val request = new GetTransactionsRequest(
+        ledgerId = unwrappedLedgerId,
+        begin = Option(ledgerBegin),
+        end = None,
+        filter = Some(
+          new TransactionFilter(
+            Map(party -> new Filters)
+          )
+        ),
+      )
+      stub(TransactionServiceGrpc.stub(channel), token)
+        .getTransactions(request, observer)
+      observer
+    }
+
+    val canActAsAlice = Right(Right.Kind.CanActAs(Right.CanActAs(partyAlice)))
+    for {
+      (userAlice, tokenAlice) <- createUserByAdmin(
+        userId = userIdAlice,
+        rights = Vector(canActAsAlice),
+      )
+      applicationId = userAlice.id
+      submitAndWaitF = () =>
+        submitAndWait(token = tokenAlice, party = partyAlice, applicationId = applicationId)
+      _ <- submitAndWaitF()
+      streamObserver = observeTransactionsStream(tokenAlice, partyAlice)
+      _ <- submitAndWaitF()
+      // Making a change to the user Alice
+      _ <- grantUserRightsByAdmin(
+        userId = userIdAlice,
+        Right(Right.Kind.CanActAs(Right.CanActAs(UUID.randomUUID().toString))),
+      )
+      _ = Thread.sleep(UserManagementCacheExpiryInSeconds.toLong * 1000)
+      //
+      _ <- submitAndWaitF()
+      _ <- submitAndWaitF()
+      _ <- submitAndWaitF()
```

Comment:
Remove these, as new elements shouldn't be required to abort the stream.

Reply:
Done